components/Ink/src/private/MDCSimpleInkLayer.m

```diff
@@ -0,0 +1,156 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCSimpleInkLayer.h"
+#import "MaterialMath.h"
+
+static NSString *const kMDCSimpleInkLayerOpacityString = @"opacity";
+static NSString *const kMDCSimpleInkLayerPositionString = @"position";
+static NSString *const kMDCSimpleInkLayerScaleString = @"transform.scale";
+
+@implementation MDCSimpleInkLayer
+
+- (instancetype)initWithLayer:(id)layer {
+  self = [super initWithLayer:layer];
+  if (self) {
+    _initialRadius = 0;
+    _finalRadius = 0;
+    _inkColor = [UIColor colorWithWhite:0 alpha:0.08f];
+    _startAnimationActive = NO;
+    _endAnimationDelay = 0;
+  }
+  return self;
+}
+
+- (void)setFrame:(CGRect)frame {
+  [super setFrame:frame];
+  self.initialRadius = (CGFloat)(MDCHypot(CGRectGetHeight(frame), CGRectGetWidth(frame)) / 2 * 0.6);
+  self.finalRadius = (CGFloat)(MDCHypot(CGRectGetHeight(frame), CGRectGetWidth(frame)) / 2 + 10.f);
+}
+
+- (void)startAnimationAtPoint:(CGPoint)point {
+  CGFloat radius = self.finalRadius;
+  CGRect ovalRect = CGRectMake(CGRectGetWidth(self.bounds) / 2 - radius,
+                               CGRectGetHeight(self.bounds) / 2 - radius,
+                               radius * 2,
+                               radius * 2);
+  UIBezierPath *circlePath = [UIBezierPath bezierPathWithOvalInRect:ovalRect];
+  self.path = circlePath.CGPath;
+  self.fillColor = self.inkColor.CGColor;
+  self.opacity = 0;
+  self.position = point;
+  [self startAnimation:point];
+}
+
+- (void)startAnimation:(CGPoint)point {
```

Comment:
We have both startAnimationAtPoint:(CGPoint)point and startAnimation:(CGPoint)point.
This is confusing.  Do we need them both?  If so I would add some comments explaining why.
Naming is hard.